File: python/anagrams.py
import collections

def groupAnagrams(strs):
    anagrams = collections.defaultdict(list);
    for word in strs:
        anagrams[''.join(sorted(word))].append(word)
    return anagrams.values()

print(groupAnagrams(["eat", "tea", "tan", "ate", "nat", "bat"]))

File: php/anagrams.php
<?php

function groupAnagrams(array $strList)
{
    $map = [];
    foreach($strList as $str) {
        $tmp = str_split($str);
        sort($tmp);
        $key = implode('', $tmp);

        if (!array_key_exists($key, $map)) {
            $map[$key] = [];
        }

        $map[$key][] = $str;
    }

    return array_values($map);
}

var_dump(groupAnagrams(["eat", "tea", "tan", "ate", "nat", "bat"]));

